lib/authentication/view/edit_profile_page.dart

```diff
@@ -213,48 +214,45 @@ class _EditProfilePageState extends State<EditProfilePage> {
     );
   }
 
-  Future<void> getImage() async {
-    final Uint8List image = await StorageProvider.getImage();
-    setState(() {
-      if (image != null) {
-        this.image = image;
-      } else {
-        AppToast.show(S.of(context).errorImage);
-      }
-    });
-  }
-
-  Widget loadImage() {
+  Widget buildEditableAvatar(BuildContext context) {
     final AuthProvider authProvider = Provider.of<AuthProvider>(context);
-    if (image != null) {
-      return CircleImage(
-        circleSize: 150,
-        image: MemoryImage(image),
-        enableOverlay: true,
-        overlayIcon: const Icon(Icons.edit),
-      );
+    ImageProvider<dynamic> image;
+    if (imageWidget != null) {
+      image = imageWidget;
+    } else {
+      authProvider.getProfilePicture(context).then((value) => {
+            image = NetworkImage(value),
+            setState(() {
+              imageWidget = image;
+            })
+          });
     }
```

Comment:
You could actually call this directly from `initState`. It should make it marginally faster, since we'd fetch the profile picture when we start building rather than after.